cpp/oneapi/dal/table/backend/csr_table_impl.hpp

```diff
@@ -31,44 +31,58 @@ class csr_table_impl : public detail::csr_table_template<csr_table_impl>,
             : col_count_(0),
               row_count_(0),
               layout_(data_layout::row_major),
-              csr_indexing_(detail::csr_indexing::one_based) {}
+              indexing_(sparse_indexing::one_based) {}
 
-    csr_table_impl(std::int64_t column_count,
-                   std::int64_t row_count,
-                   const array<byte_t>& data,
+    csr_table_impl(const array<byte_t>& data,
                    const array<std::int64_t>& column_indices,
-                   const array<std::int64_t>& row_indices,
+                   const array<std::int64_t>& row_offsets,
+                   std::int64_t column_count,
                    data_type dtype,
-                   detail::csr_indexing indexing)
+                   sparse_indexing indexing)
             : meta_(create_metadata(column_count, dtype)),
               data_(data),
               column_indices_(column_indices),
-              row_indices_(row_indices),
+              row_offsets_(row_offsets),
               col_count_(column_count),
-              row_count_(row_count),
               layout_(data_layout::row_major),
-              csr_indexing_(indexing) {
+              indexing_(indexing) {
         using error_msg = dal::detail::error_messages;
 
-        if (row_count <= 0) {
-            throw dal::domain_error(error_msg::rc_leq_zero());
-        }
-
         if (column_count <= 0) {
             throw dal::domain_error(error_msg::cc_leq_zero());
         }
 
-        if (indexing != detail::csr_indexing::one_based) {
-            throw dal::domain_error(detail::error_messages::zero_based_indexing_is_not_supported());
-        }
-
-        const std::int64_t element_count = row_indices_[row_count] - 1;
+        row_count_ = (row_offsets.get_count() ? row_offsets.get_count() - 1 : 0);
+        const std::int64_t element_count = row_offsets_[row_count_] - row_offsets_[0];
         const std::int64_t dtype_size = detail::get_data_type_size(dtype);
 
         detail::check_mul_overflow(element_count, dtype_size);
         if (data.get_count() != element_count * dtype_size) {
             throw dal::domain_error(error_msg::invalid_data_block_size());
         }
+
+        for (std::int64_t i = 1; i <= row_count_; i++) {
```

Comment:
```suggestion
        for (std::int64_t i = 1; i < (row_count_ + 1); ++i) {
```

Reply:
There would be needed an additional overflow check due to this +1. I'd better leave `<=`.